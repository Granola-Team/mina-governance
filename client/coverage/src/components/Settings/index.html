
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/components/Settings</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> src/components/Settings</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">54.05% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>100/185</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>9/9</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">35.71% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>5/14</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">54.05% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>100/185</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line medium'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="Settings.tsx"><a href="Settings.tsx.html">Settings.tsx</a></td>
	<td data-value="96.42" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 96%"></div><div class="cover-empty" style="width: 4%"></div></div>
	</td>
	<td data-value="96.42" class="pct high">96.42%</td>
	<td data-value="56" class="abs high">54/56</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="66.66" class="pct medium">66.66%</td>
	<td data-value="3" class="abs medium">2/3</td>
	<td data-value="96.42" class="pct high">96.42%</td>
	<td data-value="56" class="abs high">54/56</td>
	</tr>

<tr>
	<td class="file low" data-value="SettingsButton.tsx"><a href="SettingsButton.tsx.html">SettingsButton.tsx</a></td>
	<td data-value="20.68" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 20%"></div><div class="cover-empty" style="width: 80%"></div></div>
	</td>
	<td data-value="20.68" class="pct low">20.68%</td>
	<td data-value="29" class="abs low">6/29</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="20.68" class="pct low">20.68%</td>
	<td data-value="29" class="abs low">6/29</td>
	</tr>

<tr>
	<td class="file high" data-value="SettingsControl.tsx"><a href="SettingsControl.tsx.html">SettingsControl.tsx</a></td>
	<td data-value="90.9" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 90%"></div><div class="cover-empty" style="width: 10%"></div></div>
	</td>
	<td data-value="90.9" class="pct high">90.9%</td>
	<td data-value="22" class="abs high">20/22</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="66.66" class="pct medium">66.66%</td>
	<td data-value="3" class="abs medium">2/3</td>
	<td data-value="90.9" class="pct high">90.9%</td>
	<td data-value="22" class="abs high">20/22</td>
	</tr>

<tr>
	<td class="file low" data-value="SettingsDarkMode.tsx"><a href="SettingsDarkMode.tsx.html">SettingsDarkMode.tsx</a></td>
	<td data-value="11.76" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 11%"></div><div class="cover-empty" style="width: 89%"></div></div>
	</td>
	<td data-value="11.76" class="pct low">11.76%</td>
	<td data-value="51" class="abs low">6/51</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="11.76" class="pct low">11.76%</td>
	<td data-value="51" class="abs low">6/51</td>
	</tr>

<tr>
	<td class="file medium" data-value="SettingsMenu.tsx"><a href="SettingsMenu.tsx.html">SettingsMenu.tsx</a></td>
	<td data-value="53.84" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 53%"></div><div class="cover-empty" style="width: 47%"></div></div>
	</td>
	<td data-value="53.84" class="pct medium">53.84%</td>
	<td data-value="13" class="abs medium">7/13</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="53.84" class="pct medium">53.84%</td>
	<td data-value="13" class="abs medium">7/13</td>
	</tr>

<tr>
	<td class="file low" data-value="SettingsNetwork.tsx"><a href="SettingsNetwork.tsx.html">SettingsNetwork.tsx</a></td>
	<td data-value="41.66" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 41%"></div><div class="cover-empty" style="width: 59%"></div></div>
	</td>
	<td data-value="41.66" class="pct low">41.66%</td>
	<td data-value="12" class="abs low">5/12</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="41.66" class="pct low">41.66%</td>
	<td data-value="12" class="abs low">5/12</td>
	</tr>

<tr>
	<td class="file high" data-value="index.ts"><a href="index.ts.html">index.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2022-08-04T12:48:05.775Z
            </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    