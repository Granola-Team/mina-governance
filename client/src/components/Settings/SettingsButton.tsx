import React from "react";

import type { SettingsButtonProps } from "@/types";
import { useAppStore } from "@/App.store";

export const SettingsButton: React.FC<SettingsButtonProps> = ({ title }) => {
  const { network, setNetwork } = useAppStore((state) => state);
  const active = title === network;

  return (
    <div className="flex items-center justify-center w-full">
      <button
        type="button"
        disabled={active}
        onClick={() => {
          setNetwork(title);
        }}
        className={`w-full h-full py-3 transition-all duration-200 hover:opacity-80 rounded-lg ${
          active
            ? "border-2 border-green-8 bg-green-6"
            : "border-2 border-gray-8"
        }`}
      >
        <span className="font-medium">{title}</span>
      </button>
    </div>
  );
};
