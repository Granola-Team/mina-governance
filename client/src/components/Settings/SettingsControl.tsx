import React from "react";
import shallow from "zustand/shallow";

import { CogIcon } from "@heroicons/react/solid";

import { useAppStore } from "@/App.store";

export const SettingsControl = () => {
  const { settingsActive, setSettingsState } = useAppStore(
    (state) => ({
      settingsActive: state.settingsActive,
      setSettingsState: state.setSettingsState,
    }),
    shallow,
  );

  return (
    <button
      type="button"
      aria-describedby="false"
      onClick={() => {
        setSettingsState(!settingsActive);
      }}
    >
      <div className="group px-2 md:px-3 py-1.5 hover:bg-gray-4 transition-all duration-200 rounded-md">
        <CogIcon className="w-[25px] h-[25px] text-gray-9 group-hover:text-gray-12 transition-all duration-200" />
      </div>
    </button>
  );
};
