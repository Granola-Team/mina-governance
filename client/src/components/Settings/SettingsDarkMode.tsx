import React from "react";
import shallow from "zustand/shallow";

import { useAppStore } from "@/App.store";
import { changeTheme } from "@/utils/theme";

export const SettingsDarkMode = () => {
  const { darkMode, setDarkMode } = useAppStore(
    (state) => ({ darkMode: state.darkMode, setDarkMode: state.setDarkMode }),
    shallow,
  );

  const handleToggle = (e: React.ChangeEvent<HTMLInputElement>) => {
    setDarkMode(e.target.checked);
    changeTheme(darkMode);
  };

  return (
    <div className="flex items-center justify-between">
      <div className="flex flex-col">
        <span className="font-semibold">Dark Mode</span>
        <span className="mt-1 text-sm text-gray-11">
          Enable/Disable UID-Mode
        </span>
      </div>
      <label
        htmlFor="theme-toggle"
        className="flex items-center cursor-pointer"
      >
        <div className="relative">
          <input
            checked={darkMode}
            onChange={handleToggle}
            type="checkbox"
            id="theme-toggle"
            className="sr-only"
            data-testid="toggle"
          />
          <div
            className={`rounded-full p-[2px] ${
              darkMode ? "bg-gradient animate-gradient" : ""
            }`}
          >
            <div className="block h-[22px] bg-gray-8 rounded-full w-14" />
          </div>
          <div
            className={`absolute top-0 w-[26px] h-[26px] transition bg-gray-9 rounded-full dot ${
              !darkMode ? "border border-gray-9" : ""
            }`}
          />
        </div>
      </label>
    </div>
  );
};
