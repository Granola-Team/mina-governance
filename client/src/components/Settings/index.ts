export * from "./Settings";
export * from "./SettingsControl";
