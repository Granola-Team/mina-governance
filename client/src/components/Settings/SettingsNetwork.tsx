import React from "react";

import { SettingsButton } from "./SettingsButton";

export const SettingsNetwork = () => {
  return (
    <div className="flex items-center justify-between gap-4">
      <SettingsButton title="Mainnet" />
      <SettingsButton title="Devnet" />
    </div>
  );
};
