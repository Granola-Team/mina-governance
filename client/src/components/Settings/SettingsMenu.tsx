import React from "react";

import { SettingsDarkMode } from "./SettingsDarkMode";

export const SettingsMenu = () => {
  return (
    <div className="flex flex-col gap-4 pt-4">
      <SettingsDarkMode />
    </div>
  );
};
