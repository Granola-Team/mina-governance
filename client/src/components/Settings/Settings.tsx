import React, { Fragment } from "react";
import { Dialog, Transition } from "@headlessui/react";
import { AdjustmentsIcon, XIcon } from "@heroicons/react/outline";

import { useAppStore } from "@/store/app.store";
import { SettingsMenu } from "./SettingsMenu";

export const Settings = () => {
  const settingsActive = useAppStore((state) => state.settingsActive);
  const setSettingsState = useAppStore((state) => state.setSettingsState);

  return (
    <Transition.Root show={settingsActive} as={Fragment}>
      <Dialog
        as="div"
        className="relative z-10"
        onClose={() => {
          setSettingsState(false);
        }}
      >
        <Transition.Child
          as={Fragment}
          enter="ease-out duration-300"
          enterFrom="opacity-0"
          enterTo="opacity-100"
          leave="ease-in duration-200"
          leaveFrom="opacity-100"
          leaveTo="opacity-0"
        >
          <div className="fixed inset-0 transition-opacity" />
        </Transition.Child>

        <div className="fixed left-0 right-0 z-10 overflow-y-auto">
          <div className="flex items-end justify-center min-h-full p-4 text-center sm:items-center sm:p-0">
            <Transition.Child
              as={Fragment}
              enter="ease-out duration-300"
              enterFrom="opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95"
              enterTo="opacity-100 translate-y-0 sm:scale-100"
              leave="ease-in duration-200"
              leaveFrom="opacity-100 translate-y-0 sm:scale-100"
              leaveTo="opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95"
            >
              <Dialog.Panel className="relative w-full overflow-hidden text-left border rounded-lg shadow-lg border-gray-7 sm:my-8 sm:max-w-lg">
                <div className="px-4 pt-5 pb-4 bg-grayA-2 sm:p-6 sm:pb-4">
                  <div className="pb-3 border-b sm:flex sm:items-center border-b-grayA-5">
                    <div className="flex items-center justify-center flex-shrink-0 w-12 h-12 mx-auto rounded-full bg-grayA-5 sm:mx-0 sm:h-10 sm:w-10">
                      <AdjustmentsIcon
                        className="w-6 h-6 text-gray-12"
                        aria-hidden="true"
                      />
                    </div>
                    <div className="flex items-center justify-between w-full mt-3 text-center sm:mt-0 sm:ml-4 sm:text-left">
                      <Dialog.Title
                        as="h3"
                        className="text-xl font-semibold leading-6"
                      >
                        OSC Controls
                      </Dialog.Title>
                      <XIcon
                        onClick={() => {
                          setSettingsState(false);
                        }}
                        className="w-6 h-6 cursor-pointer text-gray-12"
                        aria-hidden="true"
                      />
                    </div>
                  </div>
                  <SettingsMenu />
                </div>
              </Dialog.Panel>
            </Transition.Child>
          </div>
        </div>
      </Dialog>
    </Transition.Root>
  );
};
