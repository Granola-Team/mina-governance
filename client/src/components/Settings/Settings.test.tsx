import "@testing-library/jest-dom";
import { fireEvent, render } from "@testing-library/react";
import { vi } from "vitest";
import { Settings } from "./Settings";
import { SettingsControl } from "./SettingsControl";
import { BrowserRouter as Router } from "react-router-dom";

describe("Settings and Theme Tests", () => {
    beforeEach(() => {
        const mock = vi.fn();

        mock.mockReturnValue({
            observe: () => null,
            unobserve: () => null,
            disconnect: () => null,
    });

        window.IntersectionObserver = mock;
    });

    test("should render", () => {
        expect(
            render(
                <Router>
                    <Settings />
                </Router>,
            ),
        ).toMatchSnapshot();
    });

    test("Search icon visible", async () => {
        const rendered = render(
          <Router>
            <SettingsControl />
          </Router>,
        );
        const button = rendered.getByTestId("settings-control-btn");
        expect(button).toBeInTheDocument();
    });

    test("Click Settings icon-button for modal pop-up", async () => {
        const rendered = render(
          <Router>
            <SettingsControl />
            <Settings />
          </Router>,
        );
        const button = rendered.getByTestId("settings-control-btn");
        expect(button).toBeInTheDocument();
        fireEvent.click(button);
        expect(rendered.getByText("OSC Controls")).toBeInTheDocument();
    });
});
