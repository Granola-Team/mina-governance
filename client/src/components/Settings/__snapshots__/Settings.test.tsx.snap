// Vitest Snapshot v1

exports[`Settings and Theme Tests > should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`SettingsButton is rendered 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    class="theme-dark"
  >
    <div>
      <div
        class="flex items-center justify-center w-full"
      >
        <button
          class="w-full h-full py-3 transition-all duration-200 hover:opacity-80 rounded-lg border-2 border-gray-8"
          type="button"
        >
          <span
            class="font-medium"
          >
            Devnet
          </span>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex items-center justify-center w-full"
    >
      <button
        class="w-full h-full py-3 transition-all duration-200 hover:opacity-80 rounded-lg border-2 border-gray-8"
        type="button"
      >
        <span
          class="font-medium"
        >
          Devnet
        </span>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`SettingsNetwork is rendered 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    class="theme-dark"
  >
    <div>
      <div
        class="flex items-center justify-between gap-4"
      >
        <div
          class="flex items-center justify-center w-full"
        >
          <button
            class="w-full h-full py-3 transition-all duration-200 hover:opacity-80 rounded-lg border-2 border-gray-8"
            type="button"
          >
            <span
              class="font-medium"
            >
              Mainnet
            </span>
          </button>
        </div>
        <div
          class="flex items-center justify-center w-full"
        >
          <button
            class="w-full h-full py-3 transition-all duration-200 hover:opacity-80 rounded-lg border-2 border-gray-8"
            type="button"
          >
            <span
              class="font-medium"
            >
              Devnet
            </span>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex items-center justify-between gap-4"
    >
      <div
        class="flex items-center justify-center w-full"
      >
        <button
          class="w-full h-full py-3 transition-all duration-200 hover:opacity-80 rounded-lg border-2 border-gray-8"
          type="button"
        >
          <span
            class="font-medium"
          >
            Mainnet
          </span>
        </button>
      </div>
      <div
        class="flex items-center justify-center w-full"
      >
        <button
          class="w-full h-full py-3 transition-all duration-200 hover:opacity-80 rounded-lg border-2 border-gray-8"
          type="button"
        >
          <span
            class="font-medium"
          >
            Devnet
          </span>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
